purchasables.md

```diff
@@ -1,14 +1,132 @@
+  "contentId": "23",
+  "enabled": "1",
+  "archived": "0",
+  "siteId": "1",
+  "title": "A New Toga",
+  "slug": "ant-001",
+  "uri": null,
+  "dateCreated": "2020-06-29T09:12:02-07:00",
+  "dateUpdated": "2020-06-29T09:12:03-07:00",
+  "dateDeleted": null,
+  "trashed": false,
+  "propagateAll": false,
+  "newSiteIds": [],
+  "resaving": false,
+  "duplicateOf": null,
+  "previewing": false,
+  "hardDelete": false,
+  "ref": null,
+  "status": "enabled",
+  "structureId": null,
+  "url": "https://foo.dev/shop/products/ant-001?variant=30",
+  "isAvailable": true,
+  "isPromotable": true,
+  "shippingCategoryId": 1,
+  "taxCategoryId": 1,
+  "onSale": false,
+  "cpEditUrl": "#",
+  "product": {
+    "postDate": "2020-06-29T09:12:02-07:00",
+    "expiryDate": null,
+    "typeId": "1",
+    "taxCategoryId": "1",
+    "shippingCategoryId": "1",
+    "promotable": "1",
+    "freeShipping": null,
+    "enabled": "1",
+    "availableForPurchase": true,
+    "defaultVariantId": "30",
+    "defaultSku": "ANT-001",
+    "defaultPrice": "20.0000",
+    "defaultHeight": "0.0000",
+    "defaultLength": "0.0000",
+    "defaultWidth": "0.0000",
+    "defaultWeight": "0.0000",
+    "taxCategory": null,
+    "name": null,
+    "id": "29",
+    "tempId": null,
+    "draftId": null,
+    "revisionId": null,
+    "uid": "71a2efe8-163f-4014-9ffc-b8272d469a7e",
+    "fieldLayoutId": null,
+    "contentId": "22",
+    "archived": "0",
+    "siteId": "1",
+    "title": "A New Toga",
+    "slug": "ant-001",
+    "uri": "shop/products/ant-001",
+    "dateCreated": "2020-06-29T09:12:02-07:00",
+    "dateUpdated": "2020-06-29T09:12:02-07:00",
+    "dateDeleted": null,
+    "trashed": false,
+    "propagateAll": false,
+    "newSiteIds": [],
+    "resaving": false,
+    "duplicateOf": null,
+    "previewing": false,
+    "hardDelete": false,
+    "ref": null,
+    "status": "live",
+    "structureId": null,
+    "url": "https://foo.dev/shop/products/ant-001"
+  },
+  "description": "A New Toga",
+  "purchasableId": "30",
+  "options": [],
+  "sales": []
+}
+```
 
-A purchasable:
+:::
 
-- is an [element type](https://docs.craftcms.com/v3/extend/element-types.html)
-- implements [`craft\commerce\base\PurchasableInterface`](api:craft\commerce\base\PurchasableInterface)
-- should extend [`craft\commerce\base\Purchasable`](api:craft\commerce\base\Purchasable)
+This can be particularly helpful, for example, if `getPurchasable()` returns `null` on the line item and you need to display information from the purchasable in a customer’s order history.
 
-If you’d like to introduce your own purchasable, it’s best to extend [the base Purchasable](api:craft\commerce\base\Purchasable) because you’ll automatically get...
+::: tip
+When building your front end and displaying line items, it’s best to reference purchasable information through `getPurchasable()`, relying the `snapshot` property either as a fallback or a source of truth when historically crucial information is needed.
```

Comment:
Correct. Best to use snapshot for completed orders, if you think the purchasable could ever be deleted.

Reply:
Minor point, I think, but the purchasable could also be *changed* too, correct? To the point where as a customer I might be confused about my order history showing some detail that was different in my order.

Reply:
@mattstein correct, the purchasable could also change, so snapshot is the correct information at time of adding to the order.